src/server/scriptInfo.ts

```diff
@@ -68,37 +77,51 @@ namespace ts.server {
             this.ownFileText = false;
             this.text = undefined;
             this.lineMap = undefined;
+            this.fileSize = undefined;
         }
 
         /**
          * Set the contents as newText
          * returns true if text changed
          */
-        public reload(newText: string) {
+        public reload(newText: string): boolean {
             Debug.assert(newText !== undefined);
 
             // Reload always has fresh content
             this.pendingReloadFromDisk = false;
 
+            // We only need both this.text and this.fileSize if this.text
+            // is artificially empty because it was too large.
+            // We assume that an empty string passed to reload is not
+            // for a file that was too large to store.
+            this.fileSize = undefined;
+
             // If text changed set the text
             // This also ensures that if we had switched to version cache,
             // we are switching back to text.
             // The change to version cache will happen when needed
             // Thus avoiding the computation if there are no changes
             if (this.text !== newText) {
-                this.useText(newText);
+                this.text = newText;

```

Comment:
Why this change? Can you please keep useText?

Reply:
I'll restore it but I have to say that I find it very confusing how many ways there are to change the state of this object.

Reply:
Added a comment indicating that it is test-only.